### PR TITLE
Reworks alien facehugger incubation 

### PR DESCRIPTION
**Pull request by GDNgit**
## What Does This PR Do
Changes a Prob(3) check to advance a stage in facehugger incubation to a 70 second timer with 15% positive and negative deviation 
Removes stage 3 due to it being a carbon copy of stage 2
Removes a small bit of RNG that didn't function for whatever reason
This PR also adds a new def for math.dm which lets us calculate %s, made by @Vi3trice 
Part of #18888
## Why It's Good For The Game
Currently, facehuggers incubation is governed by the will of RNGjesus, and it can take 30 minutes or 30 seconds for incubation to advance. Removal of RNG is good.
The timer is 70 seconds (3 minutes on average) so that it will roughly take the same time for alien larva to fully mature before changing how this worked, and after. It's also far more understandable than "wooo magic numer"
There is a deviation of 15% so that RNG is still a slight factor, and there's no set-in-stone amount of time an alien embryo needs to become a larva (This might be a bit too small). 
## Testing
Compiled and ran, did stuff with facehuggers
## Changelog
:cl: GDN , Vi3trice
tweak: Facehugger incubation is now more consistent 
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/alien/special/alien_embryo.dm [hunk 1]
```diff
@@ -7,24 +7,27 @@
...
+		AttemptGrow(burst_on_success = FALSE)
 
 /obj/item/organ/internal/body_egg/alien_embryo/prepare_eat()
 	var/obj/S = ..()
 	S.reagents.add_reagent("sacid", 10)
 	return S
 
-/obj/item/organ/internal/body_egg/alien_embryo/on_life()
+/obj/item/organ/internal/body_egg/alien_embryo/on_life() //I'm gonna do something about this, I swear
+	incubation_deviation = PERCENT_OF(rand(85, 115), incubation_time_per_stage) //The actual deviation location, and where the magic happens
```
Comment (SteelSlayer) [inline code/modules/mob/living/carbon/alien/special/alien_embryo.dm:28]: Do we need recalculate this every cycle? I would move it to where the stage advances. Before/after the `last_stage_progress = world.time` line for example.

File: code/modules/mob/living/carbon/alien/special/alien_embryo.dm [hunk 2]
```diff
@@ -7,24 +7,27 @@
 	icon_state = "larva0_dead"
 	var/stage = 0
 	var/polling = FALSE
+	var/incubation_time_per_stage = 70 SECONDS //How long it takes for an alien embryo to advance a stage in it's development// Pretend this is 70 seconds it basically is I swear
+	var/incubation_deviation = 0 //The random deviation for how long the incubation period per stage will take, ranging from -15% to +15. NOTE! If you have a better name for this var, I'd love it
+	var/last_stage_progress = 0 //Used to keep track of when incubation progressed to the next stage
```
Comment (Sirryan2002) [inline code/modules/mob/living/carbon/alien/special/alien_embryo.dm:12]:
```suggestion
	///How long it takes for an alien embryo to advance a stage in it's development
	var/incubation_time_per_stage = 70 SECONDS 
	///The random deviation for how long the incubation period per stage will take ranging from -15% to +15
	var/incubation_deviation = 0  
	///Used to keep track of when incubation progressed to the next stage
	var/last_stage_progress = 0 
```

File: code/modules/mob/living/carbon/alien/special/alien_embryo.dm (hunk 1 above)
Comment (Sirryan2002) [reply code/modules/mob/living/carbon/alien/special/alien_embryo.dm:28]: I agree with steel here

File: code/modules/mob/living/carbon/alien/special/alien_embryo.dm (hunk 2 above)
Comment (GDNgit) [reply code/modules/mob/living/carbon/alien/special/alien_embryo.dm:12]: oops, lemme fix that RQ

Comment (GDNgit) [reply code/modules/mob/living/carbon/alien/special/alien_embryo.dm]: Addressed!

File: code/modules/mob/living/carbon/alien/special/alien_embryo.dm (hunk 1 above)
Comment (GDNgit) [reply code/modules/mob/living/carbon/alien/special/alien_embryo.dm]: Addressed!

**Review by SteelSlayer — APPROVED** [review]

**Pull request merged**